### DISCUSSION TITLE
How was style introduced？

### DISCUSSION THREAD

**Discussion by Angry-Sparrow** [rsuite/rsuite#1678] [Q&A] [answered]
> Recently, I am reading the source code of rsuite and trying to build a similar component library by myself.
> However, I am very confused about how to introduce the style of components?
>  I couldn't find the operation in the code.

> I would be very grateful if someone knows and gives me a reply.

**Comment by SevenOutman** [comment, marked as answer]
RSuite does not import styles into the components themselves. Styles are packed within the npm package and users import the styles in their projects. [Guide](https://rsuitejs.com/guide/usage)

**Reply by Angry-Sparrow** [reply]
I understand. Thank you very much. But I don't understand why. It seems unfriendly.

**Reply by SevenOutman** [reply]
Because RSuite is using a preprocessor-based style implementation currently. But we may move to a css-in-js approach in future if we find it bringing considerable advantage. Stay tuned!
